agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/NetworkStatsbeat.java

```diff
@@ -244,7 +242,18 @@ private static <T> void addCommonProperties(
     }
   }
 
-  private static class IntervalMetrics<T> {
+  private static class IntervalMetricsKey {
```

Comment:
you'll need to override `hashCode()` and `equals()` to use this as a map key, the easiest way to do this is using auto-value, which we already have in gradle file, but doesn't look like we're using anywhere

```
  compileOnly("com.google.auto.value:auto-value-annotations")
  annotationProcessor("com.google.auto.value:auto-value")
```

```suggestion
    @AutoValue
    private static abstract class IntervalMetricsKey {
```